openstack/networking/v2/extensions/bgp/speakers/requests.go

```diff
@@ -146,3 +146,68 @@ func RemoveBGPPeer(c *gophercloud.ServiceClient, bgpSpeakerID string, opts Remov
 	_, r.Header, r.Err = gophercloud.ParseResponse(resp, err)
 	return
 }
+
+// GetAdvertisedRoutes a.k.a. GET /v2.0/bgp-speakers/{bgp-speaker-id}/bgp-dragents
+func GetAdvertisedRoutes(c *gophercloud.ServiceClient, bgpSpeakerID string) pagination.Pager {
+	url := getAdvertisedRoutesURL(c, bgpSpeakerID)
+	return pagination.NewPager(c, url, func(r pagination.PageResult) pagination.Page {
+		return AdvertisedRoutePage{pagination.SinglePageBase(r)}
+	})
+}
+
+// AddGatewayNetworkOptsBuilder declare a function that build AddGatewayNetworkOpts into a request body.
+type AddGatewayNetworkOptsBuilder interface {
+	ToBGPSpeakerAddGatewayNetworkMap() (map[string]interface{}, error)
+}
+
+// AddGatewayNetworkOpts represents the data that would be PUT to the endpoint
+type AddGatewayNetworkOpts struct {
+	// The uuid of the network
+	NetworkID string `json:"network_id"`
+}
+
+// ToBGPSpeakerAddGatewayNetworkMap implements the function
+func (opts AddGatewayNetworkOpts) ToBGPSpeakerAddGatewayNetworkMap() (map[string]interface{}, error) {
+	return gophercloud.BuildRequestBody(opts, "")
+}
+
+// AddGatewayNetwork a.k.a. PUT /v2.0/bgp-speakers/{bgp-speaker-id}/add_gateway_network
+func AddGatewayNetwork(c *gophercloud.ServiceClient, bgpSpeakerID string, opts AddGatewayNetworkOpts) (r AddGatewayNetworkResult) {
```

Comment:
Same comment as in https://github.com/gophercloud/gophercloud/pull/2400, we should make use of the `AddGatewayNetworkOptsBuilder` interface.

Reply:
Fixed.